git-server/src/lib.rs

```diff
@@ -71,10 +70,13 @@ impl Context {
 
         tracing::debug!("Root path set to: {:?}", root_path);
 
+        let root = root_path.canonicalize()?;
+        let git_receive_hook = root.join("hooks/receive-hook");
```

Comment:
You sure this works and you don't need to do `root.join("hook").join("receive-hook")`?

Reply:
works on linux, not sure about other path types though

Reply:
Might be better to let the path module handle path separation anyway.

Reply:
Hm... changed it, but I expect it to be smart enough to handle those paths as well. (AFAIK they work on windows too, e.g. `C:\Document\proj/root/hooks/...`, not sure though)